solutions/hw5/treap.py

```diff
@@ -0,0 +1,122 @@
+from collections import MutableMapping
+from typing import Iterator, Callable, Any
+from random import randint
+
+
+class Node:
+    priority: int
+
+    def __init__(self, key, value):
+        self.key = key
+        self.value = value
+        self.left_child = None
+        self.right_child = None
+        self.priority = randint(0, 100)
+
+    def inorder(self) -> list:
+        values = []
+        if self.have_left_child():
+            values.extend(self.left_child.inorder())
+        values.append(self.value)
+        if self.have_right_child():
+            values.extend(self.right_child.inorder())
+        return values
+
+    def have_left_child(self):
+        return self.left_child is not None
+
+    def have_right_child(self):
+        return self.right_child is not None
+
+
+def split(node: Node, key):
```

Comment:
I think this function should be in the Deramida class

Reply:
Also what about type annotation for key and returned value?

Reply:
In python by default functions return None value. I think you should simplify functions and return something by default because it is a weak place and a potential place for bugs. For example:
```
if node is None:
        return None, None
if key > node.key:
    left, right = split(node.right_child, key)
    node.right_child = left
    return node, right
if key < node.key:
    left, right = split(node.left_child, key)
    node.left_child = right
    return left, node
return node.left_child, node.right_child
```